modules/node_modules/@ncigdc/uikit/withToggle.js

```diff
@@ -1,20 +1,42 @@
 import React, { Component, PropTypes } from 'react';
+import _ from 'lodash';
 
 export default function withToggle(ComposedComponent) {
   return class Toggle extends Component {
+    static propTypes = {
+      // Allow active state changes to trigger actions such as toggling relay variables
+      onActivate: PropTypes.func,
+      onDeactivate: PropTypes.func,
+      active: PropTypes.bool,
+    };
+
+    static defaultProps = {
+      onActivate: _.noop,
+      onDeactivate: _.noop,
+      active: false,
+    };
+
     constructor(props) {
       super(props);
-      this.state = { active: props.active || false };
+      this.state = { active: props.active };
     }
 
-    setActive = (active: boolean): void => this.setState({ active });
-    toggleActive = (): void => this.setState({ active: !this.state.active });
+    setActive = (active: boolean): void => {
+      this.setState({ active });
+      if (active) {
+        this.props.onActivate();
+      } else {
+        this.props.onDeactivate();
+      }
+    };
+
+    toggleActive = (): void => this.setActive(!this.state.active);
 
     render() {
       return (
         <ComposedComponent
           {...this.state}
-          {...this.props}
+          {..._.omit(this.props, _.keys(Toggle.defaultProps))}
```

Comment:
I can definitely prefer native methods if that's desired. Figured I'm using `_.omit` anyway, didn't see the benefit of using `Object.keys` here. Let's add that to our eslintrc if we prefer native methods over lodash ones

JDD thinks there's a benfit of preferring lodash functions, but then of course he would :)
https://github.com/eslint/eslint/issues/5017#issuecomment-173607411